x-pack/plugins/security/server/saved_objects/index.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IClusterClient, KibanaRequest, LegacyRequest } from '../../../../../src/core/server';
+import { SecureSavedObjectsClientWrapper } from './secure_saved_objects_client_wrapper';
+import { LegacyAPI } from '../plugin';
+import { Authorization } from '../authorization';
+import { SecurityAuditLogger } from '../audit';
+
+interface SetupSavedObjectsParams {
+  adminClusterClient: IClusterClient;
+  auditLogger: SecurityAuditLogger;
+  authz: Pick<Authorization, 'mode' | 'actions' | 'checkSavedObjectsPrivilegesWithRequest'>;
+  legacyAPI: Pick<LegacyAPI, 'savedObjects'>;
+}
+
+export function setupSavedObjects({
+  adminClusterClient,
+  auditLogger,
+  authz,
+  legacyAPI: { savedObjects },
+}: SetupSavedObjectsParams) {
+  const getKibanaRequest = (request: KibanaRequest | LegacyRequest) =>
```

Comment:
__note:__ we have a very weird situation right now, I see we're getting both `KibanaRequest` and `LegacyRequest` from various places :/

Reply:
😬 I guess that's what happens when the TS interface lets you use `any`

Reply:
yeah :disappointed: